src/SFML/Window/Unix/KeySymToUnicodeMapping.cpp

```diff
@@ -405,28 +405,31 @@ Uint32 keysymToUnicode(KeySym keysym)
         case 0x01ff: return 0x02d9; // abovedot
         case 0x01b2: return 0x02db; // ogonek
         case 0x01bd: return 0x02dd; // doubleacute
-        case 0x1ef2: return 0x0300; // combining_grave
-        case 0xfe50: return 0x0300; // dead_grave
-        case 0x1ef3: return 0x0301; // combining_acute
-        case 0xfe51: return 0x0301; // dead_acute
-        case 0xfe52: return 0x0302; // dead_circumflex
-        case 0x1e9f: return 0x0303; // combining_tilde
-        case 0xfe53: return 0x0303; // dead_tilde
-        case 0xfe54: return 0x0304; // dead_macron
-        case 0xfe55: return 0x0306; // dead_breve
-        case 0xfe56: return 0x0307; // dead_abovedot
-        case 0xfe57: return 0x0308; // dead_diaeresis
+
+        // Edited to map dead keys to non-combining characters when possible
+        case 0x1ef2: return 0x0060; // 0x0300 // combining_grave
+        case 0xfe50: return 0x0060; // 0x0300 // dead_grave
+        case 0x1ef3: return 0x00b4; // 0x0301 // combining_acute
+        case 0xfe51: return 0x00b4; // 0x0301 // dead_acute
+        case 0xfe52: return 0x005e; // 0x0302 // dead_circumflex
+        case 0x1e9f: return 0x007e; // 0x0303 // combining_tilde
+        case 0xfe53: return 0x007e; // 0x0303 // dead_tilde
+        case 0xfe54: return 0x00af; // 0x0304 // dead_macron
+        case 0xfe55: return 0x02d8; // 0x0306 // dead_breve
+        case 0xfe56: return 0x02d9; // 0x0307 // dead_abovedot
+        case 0xfe57: return 0x00a8; // 0x0308 // dead_diaeresis
         case 0x1efe: return 0x0309; // combining_hook
         case 0xfe61: return 0x0309; // dead_hook
-        case 0xfe58: return 0x030a; // dead_abovering
-        case 0xfe59: return 0x030b; // dead_doubleacute
-        case 0xfe5a: return 0x030c; // dead_caron
+        case 0xfe58: return 0x02da; // 0x030a // dead_abovering
+        case 0xfe59: return 0x02dd; // 0x030b // dead_doubleacute
+        case 0xfe5a: return 0x02c7; // 0x030c // dead_caron
         case 0xfe62: return 0x031b; // dead_horn
         case 0x1eff: return 0x0323; // combining_belowdot
         case 0xfe60: return 0x0323; // dead_belowdot
-        case 0xfe5b: return 0x0327; // dead_cedilla
-        case 0xfe5c: return 0x0328; // dead_ogonek
-        case 0xfe5d: return 0x0345; // dead_iota
+        case 0xfe5b: return 0x00b8; // 0x0327 // dead_cedilla
+        case 0xfe5c: return 0x02db; // 0x0328 // dead_ogonek
+        case 0xfe5d: return 0x037a; // 0x0345 // dead_iota
```

Comment:
Does the combine character code provide any additional information? 
Otherwise it might make more sense to remove it.
If it's useful, then we might want to change the secondary `//` as it's a bit confusing to have "comments" in comments 😄

Reply:
I think it is nice to show which values from the original table were changed. I will try to make it look better.